### PR TITLE
fix: Issue 1874 - Colorize checksec output

### PR DESCRIPTION
**Pull request by CristhianMotoche**
Hello there! I took a look at #1874 and I decided to start my first contribution to this cool project. My approach was to parse the result of `call_cmd` which the output of `checksec`. I parsed the output and added the color in a similar way to the response of `checksec`.

image

*NOTE:* Maybe there is another approach that doesn't involve parsing the command output since that is a little bit flaky. I tried making a call to `os.system` but it didn't work.

Solves issue #1874 

---

As part of [Stack Builders Inc.](https://www.stackbuilders.com) Hacktoberfest 2023 event.

### CODE REVIEW

**Review by disconnect3d — CHANGES_REQUESTED** [review]
Hi, we have `pwndbg.color.*` functions to color things - let's use them instead of reimplementing coloring :)

**Comment by CristhianMotoche** [comment]
Thanks for the tip @disconnect3d Let me know if it looks a bit better now.

File: pwndbg/commands/checksec.py [hunk 1]
```diff
@@ -1,11 +1,30 @@
...
+        .replace(":", f":{pwndbg.color.GREEN}")
+        .replace("No", f"{pwndbg.color.RED}No")
+    )
+
+
+NEW_LINE = "\n"
+
+
+def color_lines(output: str) -> str:
+    return NEW_LINE.join(list(map(color_line, output.split(NEW_LINE))))
```
Comment (disconnect3d) [inline pwndbg/commands/checksec.py]:
```suggestion
    return "\n".join(map(color_line, output.split(NEW_LINE)))
```

File: pwndbg/commands/checksec.py [hunk 2]
```diff
@@ -1,11 +1,30 @@
...
 
+def color_line(line: str) -> str:
+    return pwndbg.color.normal(
+        line.replace("*", pwndbg.color.green("*"))
+        .replace(":", f":{pwndbg.color.GREEN}")
+        .replace("No", f"{pwndbg.color.RED}No")
+    )
+
+
+NEW_LINE = "\n"
```
Comment (disconnect3d) [inline pwndbg/commands/checksec.py]:
```suggestion
```

File: pwndbg/commands/checksec.py [hunk 3]
```diff
@@ -1,11 +1,29 @@
...
+def color_line(line: str) -> str:
+    return pwndbg.color.normal(
+        line.replace("*", pwndbg.color.green("*"))
+        .replace(":", f":{pwndbg.color.GREEN}")
+        .replace("No", f"{pwndbg.color.RED}No")
+    )
+
+
+
+
```
Comment (disconnect3d) [inline pwndbg/commands/checksec.py]:
```suggestion
```

File: pwndbg/commands/checksec.py [hunk 4]
```diff
@@ -1,11 +1,28 @@
...
 
 
+def color_line(line: str) -> str:
+    return pwndbg.color.normal(
+        line.replace("*", pwndbg.color.green("*"))
+        .replace(":", f":{pwndbg.color.GREEN}")
+        .replace("No", f"{pwndbg.color.RED}No")
+    )
+
+
```
Comment (disconnect3d) [inline pwndbg/commands/checksec.py:18]:
```suggestion
```

File: pwndbg/commands/checksec.py [hunk 5]
```diff
@@ -1,11 +1,27 @@
...
+def color_lines(output: str) -> str:
+    return "\n".join(map(color_line, output.split(NEW_LINE)))
```
Comment (disconnect3d) [inline pwndbg/commands/checksec.py]:
```suggestion
    return "\n".join(map(color_line, output.split("\n")))
```

**Review by disconnect3d — APPROVED** [review]

**Comment by disconnect3d** [comment]
I'm going to merge this since this should just work. However, in the future we should just use checksec from pwntools since it is our dependency right now

**Pull request merged**